Upgrade google.golang.org/grpc dependency

yab's current version of `google.golang.org/grpc` is almost a year old: https://github.com/grpc/grpc-go/commit/32fb0ac620c32ba40a4626ddf94d90d12cce3455

The monorepo has a dependency on `yab`, therefore when I upgrade the version of `grpc-go` in monorepo, I am also upgrading this dependency for `yab`. After upgrading the dependency to  https://github.com/grpc/grpc-go/commit/501c41df7f472c740d0674ff27122f3f48c80ce7 you can no longer build `yab` (or anything in monorepo that depends on it) due to the following build error:

```
com_github_yarpc_yab/protobuf/source_reflection.go:38:3: r.InitialAddrs undefined (type *manual.Resolver has no field or method InitialAddrs)
```

It appears `grpc-go` introduced an API breaking change somewhere between https://github.com/grpc/grpc-go/commit/32fb0ac620c32ba40a4626ddf94d90d12cce3455 and https://github.com/grpc/grpc-go/commit/501c41df7f472c740d0674ff27122f3f48c80ce7.

--- a/encoding/protobuf_health_test.go
+++ b/encoding/protobuf_health_test.go
@@ -40,7 +40,7 @@ func TestProtobufHealthRequest(t *testing.T) {
 		{
 			desc:        "fail",
 			serviceName: string([]byte{0xc3, 0x28}),
-			errMsg:      "proto: invalid UTF-8 string",
+			errMsg:      "invalid UTF-8",
 		},
 	}
 

--- a/glide.lock
+++ b/glide.lock
@@ -1,6 +1,8 @@
-hash: 006acdba74cd43aabedce565abf8ae3739d68a11225ef5065c0a0a2a7047a77d
-updated: 2018-08-28T15:33:13.708606-07:00
+hash: f1a5f79e8de2c3e4cd01e48162825138799f83c93a456ea094b3423c89e20d7f
+updated: 2019-09-27T14:57:18.093679-07:00
 imports:
+- name: github.com/anmitsu/go-shlex
+  version: 648efa622239a2f6ff949fed78ee37b48d499ba4
 - name: github.com/apache/thrift
   version: b2a4d4ae21c789b689dd162deb819665567f481c
   subpackages:
@@ -9,25 +11,70 @@ imports:
   version: 3a771d992973f24aa725d07868b467d1ddfceafb
   subpackages:
   - quantile
+- name: github.com/BurntSushi/toml
+  version: 3012a1dbe2e4bd1391d42b32f0577cb7bbc7f005
 - name: github.com/cactus/go-statsd-client
   version: 5e3c483eb9321a69283f0bb517894e46c108b88b
   subpackages:
   - statsd
+  - statsd/statsdtest
 - name: github.com/casimir/xdg-go
   version: 372ccc2180dab73316615641d5617c5ed3e35529
-- name: github.com/codahale/hdrhistogram
-  version: 3a0bb77429bd3a61596f5e8a3172445844342120
 - name: github.com/davecgh/go-spew
   version: 8991bc29aa16c548c550c7ff78260e27b9ab7c73
   subpackages:
   - spew
+- name: github.com/fatih/structtag
+  version: 76ae1d6d2117609598c7d4e8f3e938145f204e8f
+- name: github.com/gogo/googleapis
+  version: b8d18e97a9a193c846d32143391f6033a9d69849
+  subpackages:
+  - google/rpc
+- name: github.com/gogo/protobuf
+  version: 0ca988a254f991240804bf9821f3450d87ccbb1b
+  subpackages:
+  - gogoproto
+  - plugin/compare
+  - plugin/defaultcheck
+  - plugin/description
+  - plugin/embedcheck
+  - plugin/enumstringer
+  - plugin/equal
+  - plugin/face
+  - plugin/gostring
+  - plugin/marshalto
+  - plugin/oneofcheck
+  - plugin/populate
+  - plugin/size
+  - plugin/stringer
+  - plugin/testgen
+  - plugin/union
+  - plugin/unmarshal
+  - proto
+  - protoc-gen-gogo/descriptor
+  - protoc-gen-gogo/generator
+  - protoc-gen-gogo/generator/internal/remap
+  - protoc-gen-gogo/grpc
+  - protoc-gen-gogo/plugin
+  - protoc-gen-gogoslick
+  - sortkeys
+  - types
+  - vanity
+  - vanity/command
+- name: github.com/gogo/status
+  version: 935308aef7372e7685e8fbee162aae8f7a7e515a
+- name: github.com/golang/mock
+  version: 9fa652df1129bef0e734c9cf9bf6dbae9ef3b9fa
+  subpackages:
+  - gomock
+  - mockgen
+  - mockgen/model
 - name: github.com/golang/protobuf
-  version: 668a607657a5d387d6333648a2d9c749761fdc69
+  version: 6c65a5562fc06764971b7c5d05c76c75e84bdbf7
   subpackages:
   - jsonpb
   - proto
   - protoc-gen-go/descriptor
-  - protoc-gen-go/plugin
   - ptypes
   - ptypes/any
   - ptypes/duration
@@ -38,32 +85,37 @@ imports:
 - name: github.com/jessevdk/go-flags
   version: 1c38ed7ad0cc3d9e66649ac398c30e45f395c4eb
 - name: github.com/jhump/protoreflect
-  version: 95c5cbbeaee7fe3c2b5ecf0a163144140dfb4d61
+  version: a84568470d8add1307a0f1d8a3afcc4dfe7c4dd0
   subpackages:
   - desc
   - desc/internal
-  - desc/protoparse
   - dynamic
+  - grpcreflect
   - internal
+- name: github.com/kisielk/errcheck
+  version: e14f8d59a22d460d56c5ee92507cd94c78fbf274
+  subpackages:
+  - internal/errcheck
 - name: github.com/matttproud/golang_protobuf_extensions
   version: c12348ce28de40eed0136aa2b644d0ee0650e56c
   subpackages:
   - pbutil
 - name: github.com/opentracing/opentracing-go
-  version: 1949ddbfd147afd4d964a9f00b24eb291e0e7c38
+  version: 659c90643e714681897ec2521c60567dd21da733
   subpackages:
   - ext
   - log
 - name: github.com/pkg/errors
-  version: 645ef00459ed84a119197bfb8d8205042c6df63d
+  version: ba968bfe8b2f7e042a574c888954fccecfa385b4
 - name: github.com/pmezard/go-difflib
   version: 792786c7400a136282c1664665ae0a8db921c6c2
   subpackages:
   - difflib
 - name: github.com/prometheus/client_golang
-  version: c5b7fccd204277076155f10851dad72b76a49317
+  version: 170205fb58decfd011f1550d4cfb737230d7ae4f
   subpackages:
   - prometheus
+  - prometheus/internal
   - prometheus/promhttp
 - name: github.com/prometheus/client_model
   version: 99fa1f4be8e564e8a6b613da7fa6f46c9edafc6c
@@ -76,28 +128,27 @@ imports:
   - internal/bitbucket.org/ww/goautoneg
   - model
 - name: github.com/prometheus/procfs
-  version: 8b1c2da0d56deffdbb9e48d4414b4e674bd8083e
+  version: 499c85531f756d1129edd26485a5f73871eeb308
   subpackages:
+  - internal/fs
   - internal/util
-  - nfs
-  - xfs
+- name: github.com/samuel/go-thrift
+  version: e8b6b52668fe9c972220addc130edf46a9b466b1
+  subpackages:
+  - parser
 - name: github.com/stretchr/testify
   version: c679ae2cc0cb27ec3293fea7e254e47386f05d69
   subpackages:
   - assert
   - require
-- name: github.com/uber-go/atomic
-  version: 1ea20fb1cbb1cc08cbd0d913a96dead89aa18289
-  subpackages:
-  - utils
 - name: github.com/uber-go/mapdecode
   version: 718b4994083e432669f44a00174c5f1bcdb1434d
   subpackages:
   - internal/mapstructure
 - name: github.com/uber-go/tally
-  version: ff17f3c43c065c3c2991f571e740eee43ea3a14a
+  version: 3332297784e46cd346ab6d9894fd4ea027dc9368
 - name: github.com/uber/jaeger-client-go
-  version: b043381d944715b469fd6b37addfd30145ca1758
+  version: d4d621cd6ed9aa5f1fc91eb0d0211d336bd6b79d
   subpackages:
   - config
   - internal/baggage
@@ -113,13 +164,14 @@ imports:
   - thrift-gen/jaeger
   - thrift-gen/sampling
   - thrift-gen/zipkincore
+  - transport
   - utils
 - name: github.com/uber/jaeger-lib
-  version: ed3a127ec5fef7ae9ea95b01b542c47fbd999ce5
+  version: a87ae9d84fb038a8d79266298970720be7c80fcd
   subpackages:
   - metrics
 - name: github.com/uber/tchannel-go
-  version: 7f5c12c66261f3ac3c5dde959d65eed59e3b63af
+  version: 5ee9e351d45f66dc690528cbf014fe5b64b23bc9
   subpackages:
   - internal/argreader
   - json
@@ -128,38 +180,59 @@ imports:
   - relay/relaytest
   - testutils
   - testutils/goroutines
+  - testutils/thriftarg2test
   - thrift
+  - thrift/arg2
   - thrift/gen-go/meta
+  - thrift/thrift-gen
   - tnet
   - tos
   - trand
   - typed
 - name: go.uber.org/atomic
-  version: 1ea20fb1cbb1cc08cbd0d913a96dead89aa18289
+  version: df976f2515e274675050de7b3f42545de80594fd
 - name: go.uber.org/multierr
   version: 3c4937480c32f4c13a875a1829af76c98ca3d40a
 - name: go.uber.org/net/metrics
-  version: 1e19de5b971489a45d178e12a0f72a78c70e300e
+  version: 05a053b545cba614bf12635803f4749dab7bd0d0
   subpackages:
   - bucket
   - push
   - tallypush
 - name: go.uber.org/thriftrw
-  version: fb439a9a7f5a388d8606aae1c1ff6654b8b67fbe
+  version: 8cb9e79c2118f1eb873714e2d02747d6ecc62954
   subpackages:
   - ast
   - compile
   - envelope
+  - gen
   - idl
   - idl/internal
+  - internal/concurrent
+  - internal/curry
+  - internal/envelope
   - internal/envelope/exception
+  - internal/frame
+  - internal/goast
+  - internal/multiplex
+  - internal/plugin
+  - internal/plugin/builtin/pluginapigen
+  - internal/process
+  - internal/semver
+  - plugin
+  - plugin/api
   - protocol
   - protocol/binary
+  - ptr
   - thriftreflect
   - version
   - wire
+- name: go.uber.org/tools
+  version: ce2550dad7144b81ae2f67dc5e55597643f6902b
+  subpackages:
+  - update-license
 - name: go.uber.org/yarpc
-  version: 57163f1e97e7bf5573c50329be0fc302e7b145ca
+  version: 719d4248bd626e9fe9b79d2ac5b637dad79740f7
   subpackages:
   - api/backoff
   - api/encoding
@@ -175,6 +248,7 @@ imports:
   - internal/config
   - internal/digester
   - internal/errorsync
+  - internal/grpcerrorcodes
   - internal/humanize
   - internal/inboundmiddleware
   - internal/interpolate
@@ -200,15 +274,19 @@ imports:
   - yarpcconfig
   - yarpcerrors
 - name: go.uber.org/zap
-  version: ff33455a0e382e8a81d14dd7c922020b6b5e7982
+  version: 27376062155ad36be76b0f12cf1572a221d3a48c
   subpackages:
   - buffer
   - internal/bufferpool
   - internal/color
   - internal/exit
   - zapcore
+- name: golang.org/x/lint
+  version: d0100b6bd8b389f0385611eb39152c4d7c3a7905
+  subpackages:
+  - golint
 - name: golang.org/x/net
-  version: aaf60122140d3fcf75376d319f0554393160eb50
+  version: c00fd9afed17cfdca9b3e1e3b8de7ef2b3f0347b
   subpackages:
   - bpf
   - context
@@ -231,43 +309,91 @@ imports:
   - transform
   - unicode/bidi
   - unicode/norm
+- name: golang.org/x/tools
+  version: 1286b2016bb195031cab5657d3244bc996b3f2dd
+  repo: https://github.com/golang/tools
+  subpackages:
+  - cmd/stringer
+  - go/ast/astutil
+  - go/buildutil
+  - go/gcexportdata
+  - go/internal/cgo
+  - go/internal/gcimporter
+  - go/internal/packagesdriver
+  - go/loader
+  - go/packages
+  - go/types/typeutil
+  - internal/fastwalk
+  - internal/gopathwalk
+  - internal/semver
 - name: google.golang.org/genproto
   version: 7bb2a897381c9c5ab2aeb8614f758d7766af68ff
   subpackages:
   - googleapis/rpc/status
-  - protobuf/api
-  - protobuf/field_mask
-  - protobuf/ptype
-  - protobuf/source_context
 - name: google.golang.org/grpc
-  version: 32fb0ac620c32ba40a4626ddf94d90d12cce3455
-  repo: https://github.com/grpc/grpc-go
+  version: f6d0f9ee430895e87ef1ceb5ac8f39725bafceef
   subpackages:
   - balancer
   - balancer/base
   - balancer/roundrobin
+  - binarylog/grpc_binarylog_v1
   - codes
   - connectivity
   - credentials
+  - credentials/internal
   - encoding
   - encoding/proto
   - grpclog
+  - health
+  - health/grpc_health_v1
   - internal
   - internal/backoff
+  - internal/balancerload
+  - internal/binarylog
   - internal/channelz
   - internal/envconfig
   - internal/grpcrand
+  - internal/grpcsync
+  - internal/syscall
   - internal/transport
   - keepalive
   - metadata
   - naming
   - peer
+  - reflection
+  - reflection/grpc_reflection_v1alpha
   - resolver
   - resolver/dns
+  - resolver/manual
   - resolver/passthrough
+  - serviceconfig
   - stats
   - status
   - tap
 - name: gopkg.in/yaml.v2
   version: 5420a8b6744d3b0345ab293f6fcba19c978f1183
+- name: honnef.co/go/tools
+  version: 466a0476246c45c74e011a03de704d72dc83da2d
+  subpackages:
+  - arg
+  - callgraph
+  - callgraph/static
+  - cmd/staticcheck
+  - config
+  - deprecated
+  - functions
+  - internal/sharedcheck
+  - lint
+  - lint/lintdsl
+  - lint/lintutil
+  - lint/lintutil/format
+  - simple
+  - ssa
+  - ssa/ssautil
+  - ssautil
+  - staticcheck
+  - staticcheck/vrp
+  - stylecheck
+  - unused
+  - version
 testImports: []

--- a/glide.yaml
+++ b/glide.yaml
@@ -24,6 +24,8 @@ import:
   version: a84568470d8add1307a0f1d8a3afcc4dfe7c4dd0
 - package: google.golang.org/grpc
   version: ^1.14
+- package: github.com/prometheus/procfs
+  version: '>=0.0.1'
 - package: golang.org/x/sys
 testImport:
 - package: github.com/apache/thrift

--- a/protobuf/source_reflection.go
+++ b/protobuf/source_reflection.go
@@ -35,7 +35,7 @@ func NewDescriptorProviderReflection(args ReflectionArgs) (DescriptorProvider, e
 		}
 		peers[i] = resolver.Address{Addr: p, Type: resolver.Backend}
 	}
-	r.InitialAddrs(peers)
+	r.InitialState(resolver.State{Addresses: peers})
 
 	conn, err := grpc.DialContext(context.Background(),
 		r.Scheme()+":///", // minimal target to dial registered host:port pairs

--- a/protobuf/source_reflection_test.go
+++ b/protobuf/source_reflection_test.go
@@ -53,31 +53,6 @@ func TestReflectionWithProtocolInPeer(t *testing.T) {
 	assert.Contains(t, err.Error(), "peer contains scheme")
 }
 
-func TestReflectionMultiplePeers(t *testing.T) {
-	listenRefuser, err := net.Listen("tcp", "127.0.0.1:0")
-	require.NoError(t, err, "failed to listen on a port")
-	defer listenRefuser.Close()
-
-	noListen, err := net.Listen("tcp", "127.0.0.1:0")
-	require.NoError(t, err, "failed to listen on a port")
-	noListen.Close()
-
-	got, err := NewDescriptorProviderReflection(ReflectionArgs{
-		Timeout: time.Second,
-		Peers:   []string{noListen.Addr().String(), listenRefuser.Addr().String()},
-	})
-
-	require.NoError(t, err)
-	require.NotNil(t, got)
-
-	go func() {
-		conn, _ := listenRefuser.Accept()
-		conn.Close()
-	}()
-	_, err = got.FindSymbol("some-symbol")
-	require.Error(t, err)
-}
-
 func TestReflectionClosedPort(t *testing.T) {
 	ln, err := net.Listen("tcp", "127.0.0.1:0")
 	require.NoError(t, err, "failed to listen on a port")

